Sources/SwiftFusion/Inference/ChordalInitialization.swift

```diff
@@ -0,0 +1,258 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinStructures
+import TensorFlow
+
+/// A BetweenFactor alternative that uses the Chordal (Frobenious) norm on rotation for Rot3
+/// Please refer to Carlone15icra (Initialization Techniques for 3D SLAM: a Survey on Rotation Estimation and its Use in Pose Graph Optimization)
+/// for explanation.
+public struct RelaxedRotationFactorRot3: LinearizableFactor
+{
+  public typealias Variables = Tuple2<Vector9, Vector9>
```

Comment:
I still think this should be Matrix3, not Vector9

Reply:
It's a limitation we have currently as the new JacobianFactor requires the tangent vector to conform to `EuclideanVectorN`, which I don't know if it's a good idea to conform `Matrix3` to it? I can also make a wrapper type `Matrix3R` which has a `Vector9` as tangent, but again, is that a good idea?

Reply:
Well, the idea is that swift will allow us to do the **right** thing, easily and safely. The right thing here is that we are optimizing over rotation matrices but relaxing the orthonormality constraints. A Matrix3 *is* in fact a 9-dimensional manifold, so why would it not work? So, yes, it's a good idea, because pretending that we're optimizing over Vector9's is *not* a good idea.

Reply:
done